### DISCUSSION TITLE
The rc-Switch nSeparationLimit problem.

### DISCUSSION THREAD

**Discussion by PepeTheFroggie** [arendst/Tasmota#10874] [Q&A] [answered]
The constant nSeparationLimit describes the minimum length of the sync signal for receive. There is only one nSeparationLimit but multiple protocols. And that is a problem. Let's focus on protocols 1 to 7:

pulselength, Sync bit (hi,lo)  nSeparation (hi/lo)
{ 350, {  1, 31 }   350, 10850 / protocol 1
{ 650, {  1, 10 }    650, 6500 / protocol 2
{ 100, { 30, 71 }   3000, 7100 / protocol 3
{ 380, {  1,  6 }    380, 2280 / protocol 4
{ 500, {  6, 14 }   3000, 7000 / protocol 5
{ 450, { 23,  1 }   10350, 450 / protocol 6 (HT6P20B)
{ 150, {  2, 62 }    300, 9300 / protocol 7 (HS2303-PT, i. e. used in AUKEY Remote)

Protocol4 is at 2280, means wont work with nSeparationLimit > 2200.
Protocol3 is at 3000 on the "wrong side", means wont work with nSeparationLimit < 3100.

There is NO nSeparationLimit able to receive protocol3 AND protocol4. 
Since the code doesent know hi or low, but only "changed", it happily triggers on protocol3 "wrong side" of sync and then fails decoding.

Issuing a different nSeparationLimit for each protocol would work but severly limit the scanning capabilities (receive and decode unknown signals). The scanning code works inside an interrupt which i think is a very questionable way of doing things anyway.
How do we solve that problem?

**Comment by barbudor** [comment, marked as answer]
This is a problem with rcSwitch library, not Tasmota
Please report to the library author

**Comment by Jason2866** [comment]
PR #14421
